File: lib/libxl-3.8.1.0/examples/fortran/vs2008/extract/extract.f90
program extract

use libxl

interface
    subroutine convert(i,buff)
    !DEC$ ATTRIBUTES C :: convert
    !DEC$ ATTRIBUTES reference :: buff
    character(*) buff
    integer i
    end subroutine
end interface

type(BookHandle) book
type(SheetHandle) sheet
type(FormatHandle) nullFormat
integer i
double precision d
character(25) buff
 
book = xlCreateBook()
if(book%point.ne.0) then
    if(xlBookLoad(book, "..\generate\example.xls").ne.0) then
        sheet = xlBookGetSheet(book, 0)
        if(sheet%point.ne.0) then
            i = xlSheetReadStr(sheet, 2, 1, nullFormat)
            call convert(i,buff)

            if(buff.ne.'') print *,buff

            d = xlSheetReadNum(sheet, 3, 1, nullFormat)
            print *,d
        end if
    end if        
    call xlBookRelease(book)
end if
    
end program extract

File: lib/libxl-3.8.1.0/examples/c/mingw/extract.c
#include <stdio.h>
#include <conio.h>
#include "libxl.h"

int main()
{
    BookHandle book = xlCreateBook();
    if(book) 
    {
        if(xlBookLoad(book, "example.xls")) 
        {
            SheetHandle sheet = xlBookGetSheet(book, 0);
            if(sheet)
            {
                double d;
                const char* s = xlSheetReadStr(sheet, 2, 1, 0);

                if(s) printf("%s\n", s);

                d = xlSheetReadNum(sheet, 3, 1, 0);
                printf("%g\n", d);
            }
        }     
       
        xlBookRelease(book);
    }

    printf("\nPress any key to exit...");
    _getch();

    return 0;
}
